Modify ws-slave to also perform BLE scanning

tomh4iiii opened the pull request:
Remove external scanner scanner.js
Add support to ws-slave to restart scanning if any context still requires it

tomh4iiii committed:
Modify ws-slave to also perform BLE scanning

Remove external scanner scanner.js
Add support to ws-slave to restart scanning if any context still requires it

arthare commented on ws-slave.js:
```diff
@@ -22,25 +14,25 @@ function assert(f, reason) {
-  if(g_scannerSocket) {
+    // something has changed
+    if (noble.state !== 'poweredOn') {
+        //console.log("noble state is not poweredOn, ingnore scanning");
```
typo in comment

arthare commented on ws-slave.js:
```diff
@@ -22,25 +14,25 @@ function assert(f, reason) {
     if(isAnyContextScanning()) {
-      console.log("telling scanner to start")
-      g_scannerSocket.send("start");
+        //console.log("some contexts want scanning, telling scanner to start");
```
You can just delete this or convert it to a real comment

arthare commented on ws-slave.js:
```diff
@@ -117,52 +109,17 @@ if(defaultScanUuid) {
-  
+  switch(msg) {
+  case 'enable-real-scanning':
+      console.log("##### got enable-real-scanning request");
```
I guess since you've managed to make it always "real scanning" mode now, we can get rid of this stuff entirely

arthare commented on ws-slave.js:
```diff
@@ -219,6 +176,9 @@ wss.on('connection', function (ws) {
     ws.removeAllListeners('open');
     ws.removeAllListeners('message');
+    if (contextsLeft <= 0) {
+      noble.stopScanning();
```
this could probably go to notifyScanRelevantEvent, since a context appearing/disappearing is scan-relevant.  It'll still end up stopping scanning, but keeps everything moving through that chokepoint for scan/don't-scan decisions.

arthare commented on ws-slave.js:
```diff
@@ -766,3 +726,15 @@ noble.on('discover', function (peripheral) {
 });
+
+noble.on('scanStop', function () {
+    //console.log("Scanning stopped");
```
Could maybe put a short comment in talking about the behaviour you discovered and why this is here.

But also: WOOOOOO YOU FIXED IT.

tomh4iiii committed:
Update ws-slave changes per comments in PR

Fix comments and typos
Add documentation for the rational behind restarting scanning in the scanStop handler
Modify close handler to call notifyScanRelevantEvent to check if scanning is still required
